apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts

```diff
@@ -16,27 +16,38 @@ import {
   EinstaklingarGetForsjaRequest,
 } from '@island.is/clients/national-registry-v2'
 import environment from '../../../environments/environment'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
 
 @Injectable()
 export class UserService {
   constructor(
     private readonly flightService: FlightService,
     private readonly nationalRegistryService: NationalRegistryService,
     private readonly nationalRegistryIndividualsApi: EinstaklingarApi,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
   ) {}
 
-  async getRelations(authUser: AuthUser): Promise<string[]> {
-    const relations = await this.nationalRegistryIndividualsApi
-      .withMiddleware(
-        new AuthMiddleware(
-          authUser,
-          environment.nationalRegistry
-            .authMiddlewareOptions as AuthMiddlewareOptions,
-        ),
+  async getRelations(authUser: AuthUser): Promise<Array<string> | null> {
+    let relations: string[] = []
+    try {
+      relations = await this.nationalRegistryIndividualsApi
+        .withMiddleware(
+          new AuthMiddleware(
+            authUser,
+            environment.nationalRegistry
+              .authMiddlewareOptions as AuthMiddlewareOptions,
+          ),
+        )
+        .einstaklingarGetForsja(<EinstaklingarGetForsjaRequest>{
+          id: authUser.nationalId,
+        })
+    } catch (e) {
+      this.logger.error(
+        'NatReg xRoad forsja: ' + e.name + ' - ' + e.message + ' - ' + e.url,
```

Comment:
Why catch, log, return null, then throw a different error?

It's better to let original errors bottle up so they can be logged and handled consistently. It also saves a lot of boilerplate error handling code all over the place, which might be spotty.

E.g. this log line might miss some important properties from the error object, and it doesn't contribute much which is not apparent from the original stack trace or other logs.

Reply:
So this whole catch and and the null logic on the other side should be removed IMO.